### PR TITLE
Remove usage of the deprecated web_url* fields

### PR DESCRIPTION
**Pull request by ekzhang**
Main change here is that `FunctionCreateResponse.web_url` and `FunctionCreateResponse.web_url_info` have been deprecated for a while, but I noticed we were actually still using them, so I'm removing their usage and moving up the deprecation comment to version 0.51.

### CODE REVIEW

File: modal/client.py [hunk 1]
```diff
@@ -213,30 +213,30 @@ async def from_env(cls, _override_config=None) -> "_Client":
...
-                            "Token missing. Could not authenticate client. "
-                            "If you have token credentials, see modal.com/docs/reference/modal.config for setup help. "
-                            "If you are a new user, register an account at modal.com, then run `modal token new`."
-                        )
-                        raise AuthError(creds_missing_msg)
-                    else:
-                        raise
-                cls._client_from_env = client
-                return client
+            client = _Client(server_url, client_type, credentials)
```
Comment (erikbern) [inline modal/client.py:223]: Is this an unrelated change?

Comment (ekzhang) [reply modal/client.py:223]: Yeah it's unrelated, was moving things around while testing perf

**Review by erikbern — APPROVED** [review]
